apis/v1alpha1/collector_webhook.go

```diff
@@ -29,11 +30,19 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
 	ta "github.com/open-telemetry/opentelemetry-operator/internal/manifests/targetallocator/adapters"
 	"github.com/open-telemetry/opentelemetry-operator/pkg/featuregate"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/rbac"
 )
 
 var (
 	_ admission.CustomValidator = &CollectorWebhook{}
 	_ admission.CustomDefaulter = &CollectorWebhook{}
+
+	// targetAllocatorNamespaceRes is the resource attributes required for the newest target allocator capability
+	targetAllocatorNamespaceRes = &v1.ResourceAttributes{
+		Namespace: "",
+		Verb:      "list",
+		Resource:  "namespaces",
+	}
```

Comment:
How about using the full RBAC required for Prometheus CR here?

Reply:
yeah! i expected this comment but before I did that work, wanted to be sure that this overall approach made sense. Thank you :bow: